### ISSUE TITLE
ERROR! A worker was found in a dead state Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by micafer** [ansible/ansible#35895]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
core

##### ANSIBLE VERSION
Tested in two environments:
CentOS 7:
```
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Nov 20 2015, 02:00:19) [GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
```
And Ubuntu 16.04:
```
ansible 2.4.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

##### CONFIGURATION
On CentOS 7 env:
ansible-config dump --only-changed
```
DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/etc/ansible/roles', u'/usr/share/ansible/roles']
```
On Ubuntu 16.04 env:
ansible-config dump --only-changed
```

```

##### OS / ENVIRONMENT
Tested on plan CentOS 7 and Ubuntu 16.04.

##### SUMMARY
In case of using a task like that:
```
- pip: name=cryptography extra_args="-I"
```
The next task will fail with this error message, chrasing the Python interpreter.
```
ERROR! A worker was found in a dead state
Segmentation fault (core dumped)
```
I don't know if it is related with issues: #31908, #32554.

##### STEPS TO REPRODUCE

Ansible has been installed using the yum and apt tools in both O.S. (not pip).
Create a playbook like that and execute the ansible-playbook command with it:

```yaml
---
- hosts: localhost
  become: yes
  become_method: sudo
  tasks:
    - pip: name=cryptography extra_args="-I"
    - debug: msg="TEST"
```
##### EXPECTED RESULTS
```
PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [pip] *********************************************************************
changed: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "TEST"
}

PLAY RECAP *********************************************************************
localhost                  : ok=3    changed=1    unreachable=0    failed=0   
```
##### ACTUAL RESULTS

```
ansible-playbook 2.4.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
Parsed /etc/ansible/hosts inventory source with ini plugin
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost
does not match 'all'

Loading callback plugin default of type stdout, v2.0 from /usr/lib/python2.7/dist-packages/ansible/plugins/callback/default.pyc

PLAYBOOK: test.yml **************************************************************************************
1 plays in test.yml

PLAY [localhost] ****************************************************************************************

TASK [Gathering Facts] **********************************************************************************
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/basic.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/namespace.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/ansible_collector.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/default_collectors.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/_text.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/parsing/convert_bool.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/parsing/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/pycompat24.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/six/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/compat.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/timeout.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/collector.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/freebsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/openbsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/base.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/dragonfly.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/dragonfly.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/aix.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/aix.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/linux.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/platform.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/hpux.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/user.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/hurd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/other/facter.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/sunos.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/hpux.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/distribution.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/freebsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/openbsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/darwin.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/linux.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/sunos.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/sunos.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/apparmor.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/openbsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/local.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/other/ohai.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/other/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/service_mgr.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/hurd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/hpux.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/fips.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/selinux.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/pkg_mgr.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/dns.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/base.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/darwin.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/dragonfly.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/ssh_pub_keys.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/lsb.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/date_time.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/netbsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/base.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/cmdline.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/freebsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/netbsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/hardware/netbsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/caps.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/env.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/linux.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/system/python.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/network/generic_bsd.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/virtual/sysctl.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/utils.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/facts/sysctl.py
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/system/setup.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1518087120.74-253765421063694 `" && echo ansible-tmp-1518087120.74-253765421063694="` echo /root/.ansible/tmp/ansible-tmp-1518087120.74-253765421063694 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpONwSLt TO /root/.ansible/tmp/ansible-tmp-1518087120.74-253765421063694/setup.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1518087120.74-253765421063694/ /root/.ansible/tmp/ansible-tmp-1518087120.74-253765421063694/setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1518087120.74-253765421063694/setup.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1518087120.74-253765421063694/" > /dev/null 2>&1 && sleep 0'
ok: [localhost]
META: ran handlers

TASK [pip] **********************************************************************************************
task path: /test.yml:6
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/_text.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/basic.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/six/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/parsing/convert_bool.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/parsing/__init__.py
Using module_utils file /usr/lib/python2.7/dist-packages/ansible/module_utils/pycompat24.py
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/packaging/language/pip.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1518087123.5-243528921132951 `" && echo ansible-tmp-1518087123.5-243528921132951="` echo /root/.ansible/tmp/ansible-tmp-1518087123.5-243528921132951 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpaujx3Y TO /root/.ansible/tmp/ansible-tmp-1518087123.5-243528921132951/pip.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1518087123.5-243528921132951/ /root/.ansible/tmp/ansible-tmp-1518087123.5-243528921132951/pip.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1518087123.5-243528921132951/pip.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1518087123.5-243528921132951/" > /dev/null 2>&1 && sleep 0'
changed: [localhost] => {
    "changed": true, 
    "cmd": "/usr/bin/pip2 install -I cryptography", 
    "invocation": {
        "module_args": {
            "chdir": null, 
            "editable": false, 
            "executable": null, 
            "extra_args": "-I", 
            "name": [
                "cryptography"
            ], 
            "requirements": null, 
            "state": "present", 
            "umask": null, 
            "use_mirrors": true, 
            "version": null, 
            "virtualenv": null, 
            "virtualenv_command": "virtualenv", 
            "virtualenv_python": null, 
            "virtualenv_site_packages": false
        }
    }, 
    "name": [
        "cryptography"
    ], 
    "requirements": null, 
    "state": "present", 
    "stderr": "You are using pip version 8.1.1, however version 9.0.1 is available.\nYou should consider upgrading via the 'pip install --upgrade pip' command.\n", 
    "stderr_lines": [
        "You are using pip version 8.1.1, however version 9.0.1 is available.", 
        "You should consider upgrading via the 'pip install --upgrade pip' command."
    ], 
    "stdout": "Collecting cryptography\n  Using cached cryptography-2.1.4-cp27-cp27mu-manylinux1_x86_64.whl\nCollecting cffi>=1.7; platform_python_implementation != \"PyPy\" (from cryptography)\n  Using cached cffi-1.11.4-cp27-cp27mu-manylinux1_x86_64.whl\nCollecting enum34; python_version < \"3\" (from cryptography)\n  Using cached enum34-1.1.6-py2-none-any.whl\nCollecting asn1crypto>=0.21.0 (from cryptography)\n  Using cached asn1crypto-0.24.0-py2.py3-none-any.whl\nCollecting idna>=2.1 (from cryptography)\n  Using cached idna-2.6-py2.py3-none-any.whl\nCollecting six>=1.4.1 (from cryptography)\n  Using cached six-1.11.0-py2.py3-none-any.whl\nCollecting ipaddress; python_version < \"3\" (from cryptography)\nCollecting pycparser (from cffi>=1.7; platform_python_implementation != \"PyPy\"->cryptography)\nInstalling collected packages: pycparser, cffi, enum34, asn1crypto, idna, six, ipaddress, cryptography\nSuccessfully installed asn1crypto-0.24.0 cffi-1.11.4 cryptography-2.1.4 enum34-1.1.6 idna-2.6 ipaddress-1.0.19 pycparser-2.18 six-1.11.0\n", 
    "stdout_lines": [
        "Collecting cryptography", 
        "  Using cached cryptography-2.1.4-cp27-cp27mu-manylinux1_x86_64.whl", 
        "Collecting cffi>=1.7; platform_python_implementation != \"PyPy\" (from cryptography)", 
        "  Using cached cffi-1.11.4-cp27-cp27mu-manylinux1_x86_64.whl", 
        "Collecting enum34; python_version < \"3\" (from cryptography)", 
        "  Using cached enum34-1.1.6-py2-none-any.whl", 
        "Collecting asn1crypto>=0.21.0 (from cryptography)", 
        "  Using cached asn1crypto-0.24.0-py2.py3-none-any.whl", 
        "Collecting idna>=2.1 (from cryptography)", 
        "  Using cached idna-2.6-py2.py3-none-any.whl", 
        "Collecting six>=1.4.1 (from cryptography)", 
        "  Using cached six-1.11.0-py2.py3-none-any.whl", 
        "Collecting ipaddress; python_version < \"3\" (from cryptography)", 
        "Collecting pycparser (from cffi>=1.7; platform_python_implementation != \"PyPy\"->cryptography)", 
        "Installing collected packages: pycparser, cffi, enum34, asn1crypto, idna, six, ipaddress, cryptography", 
        "Successfully installed asn1crypto-0.24.0 cffi-1.11.4 cryptography-2.1.4 enum34-1.1.6 idna-2.6 ipaddress-1.0.19 pycparser-2.18 six-1.11.0"
    ], 
    "version": null, 
    "virtualenv": null
}

TASK [debug] ********************************************************************************************
task path: /test.yml:7
ERROR! A worker was found in a dead state
Segmentation fault (core dumped)

```

**Comment by micafer** [comment]
More info:
It seems that if you install ansible with pip the error does not appear.

**Comment by halcode** [comment]
seems to happen with python2.7 interpreter.

`gather_facts: false` in playbook avoids immediate error, but disables facts!

**Adding memory to container host (200MB to 512MB) solved the issue.** :palm_tree:

**Comment by peknu** [comment]
For me, this problem was related to the pip version installed by Vagrant ansible_local provisioner (8.1.2). This issue was fixed when I updated pip to latest (10.0.1) prior to executing the other pip tasks:
```yaml
- name: Install latest pip
  pip:
    name: pip
    state: latest
```

**Comment by bcoca** [comment]
These seem to reflect a resource issue, need more info to narrow down, probably dmesg and/or system logs would have the information needed.
